shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -69,6 +71,52 @@ public void setTextInputEditingState_alwaysRestartsOnAffectedDevices() {
         assertEquals(2, testImm.getRestartCount(testView));
     }
 
+    // See https://github.com/flutter/flutter/issues/31512
+    // All modern Samsung keybords are affected including non-korean languages and thus
+    // need the restart.
+    @Test
+    public void setTextInputEditingState_alwaysRestartsOnAffectedDevices2() {
```

Comment:
The tests are failing because they're running on `sdk=16`, so the `LOLLIPOP` check is blocking them. You can change them by adding `sdk=27` in the `@Config` annotation at the top.

On a second look, I'd also just replace the existing `setTextInputEditingState_alwaysRestartsOnAffectedDevices` with this one since we don't care about special casing the `ko` locale anymore. I think leaving that test around is a little confusing now.